### PR TITLE
Purge Unwanted Dependencies | Closes #2617

### LINKED ISSUE #2617: Purge Unwanted Dependencies
We want to locate and remove dependencies for the following libraries:

- blueprintjs
- deep-freeze
- final-form-arrays, final-form-calculate, + react variants (only if not used anywhere)
- immutable
- lodash
- moment, moment-duration-format, moment-timezone
- prop-types
- rememo

##### to do

- remove dependency

- find all components that utilize unwanted dependencies

- determine if any other components are using any of the above components 

- if so, then refactor component to use alternative import from AntD 

- if not, then simply remove component with unwanted dependencies

What would be ideal is if there is some way to automatically identify unused dependencies in order to remove them

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Removes unused dependencies from components
- Removes unused packages from `package.json`
- Moves all the dev related deps to `devDependencies`
- Updates all the packages to latest version
- Closes #2617

### CODE REVIEW

File: assets/src/application/ui/layout/entityList/filterBar/buttons/CardViewFilterButton.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+				icon={<AppstoreFilled />}
+				id={filterId}
+				onClick={setCardView}
+				tooltip={__('card view')}
+			/>
+		</FilterButtonWrap>
+	);
+});
+
+export default CardViewFilterButton;
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/filterBar/buttons/CardViewFilterButton.tsx]: The file name was not same as component/function name. It took me a while to figure out where the React Hook rules were being violated. Thus renamed it to reflect the component name.

File: assets/src/application/ui/layout/entityList/filterBar/buttons/FilterButtonWrap.tsx [hunk 2]
```diff
@@ -15,4 +15,5 @@ const FilterButtonWrap: React.FC<FilterButtonWrapProps> = ({ children, id, isMob
 	</div>
 );
 
-export default withViewportMatch({ isMobile: '< small' })(FilterButtonWrap);
+export default FilterButtonWrap;
+// export default withViewportMatch({ isMobile: '< small' })(FilterButtonWrap);
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/filterBar/buttons/FilterButtonWrap.tsx]: `withViewportMatch` violated React Hook Rules.

File: assets/src/application/ui/layout/entityList/filterBar/buttons/TableViewFilterButton.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+				icon={Icon.TABLE_VIEW}
+				id={filterId}
+				onClick={setTableView}
+				tooltip={__('table view')}
+			/>
+		</FilterButtonWrap>
+	);
+});
+
+export default TableViewFilterButton;
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/filterBar/buttons/TableViewFilterButton.tsx]: Same here regarding the file name.

File: assets/src/application/ui/layout/entityList/filterBar/buttons/ToggleFiltersButton.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+				icon={Icon.FILTER}
+				id={filterId}
+				onClick={toggleFilters}
+				tooltip={tooltip}
+			/>
+		</FilterButtonWrap>
+	);
+});
+
+export default ToggleFiltersButton;
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/filterBar/buttons/ToggleFiltersButton.tsx]: Same here regarding the file name

File: assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
 import React, { useState } from 'react';
 
-import CardViewFilterButton from './buttons/CardView';
+import CardViewFilterButton from './buttons/CardViewFilterButton';
 import Collapsible from './Collapsible';
 import { EntityListFilterBarProps } from './types';
 import { EntityListLegend } from '@application/ui/display';
-import TableViewFilterButton from './buttons/TableView';
-import ToggleLegendButton from './buttons/ToggleLegend';
-import ToggleFiltersButton from './buttons/ToggleFilters';
```
Comment (manzoorwanijk) [inline assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx]: When EDTR crashed with React Hook rules violation, it showed `CardView.tsx` and `TableView.tsx`, I was looking at EntityList card view components of dates and tickets 🙄

File: assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
+import TableViewFilterButton from './buttons/TableViewFilterButton';
+import ToggleLegendButton from './buttons/ToggleLegendButton';
+import ToggleFiltersButton from './buttons/ToggleFiltersButton';
```
Comment (nicolad) [inline assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx]: :+1:

File: assets/src/application/ui/display/EntityIDs/index.tsx [hunk 7]
```diff
@@ -16,21 +16,18 @@ export interface EntityIdProps {
 	guid: EntityId;
 }
 
-const EntityIDs: React.FC<EntityIdProps> = ({ dbid, guid, align = 'left' }) => {
+const EntityIDs: React.FC<EntityIdProps> = React.memo(({ dbid, guid, align = 'left' }) => {
```
Comment (nicolad) [inline assets/src/application/ui/display/EntityIDs/index.tsx]: :+1:

File: assets/src/application/ui/display/ErrorIndicator.tsx [hunk 8]
```diff
@@ -1,11 +1,7 @@
...
+import { Result } from 'antd';
+import { ResultProps } from 'antd/lib/result';
 
-const ErrorIndicator = () => (
-	<Callout>
-		<Spinner size={Spinner.SIZE_SMALL} />
-		<NonIdealState icon={'warning-sign'} />
-	</Callout>
-);
+const ErrorIndicator: React.FC<ResultProps> = (props) => <Result status='warning' {...props} />;
```
Comment (nicolad) [inline assets/src/application/ui/display/ErrorIndicator.tsx]: :taco:

File: assets/src/application/ui/input/EspressoButton/Tooltip.tsx [hunk 9]
```diff
@@ -27,10 +26,11 @@ const Tooltip: React.FC<TicketsChainedButtonProps> = ({ button, buttonText, isMo
 	return isMobile ? (
 		mobileHelpText
 	) : (
-		<DefaultTooltip title={tooltip} {...props}>
+		<DefaultTooltip title={title} {...props}>
```
Comment (tn3rb) [inline assets/src/application/ui/input/EspressoButton/Tooltip.tsx]: why this change? what was wrong with the prop being named "tooltip"?
we don't need to use the same prop names as the ANtD components!

File: assets/src/application/ui/layout/entityList/filterBar/buttons/FilterButtonWrap.tsx (hunk 2 above)
Comment (tn3rb) [reply assets/src/application/ui/layout/entityList/filterBar/buttons/FilterButtonWrap.tsx]: after this is merged to master i'll try to solve that problem using CSS

File: assets/src/application/ui/input/EspressoButton/Tooltip.tsx (hunk 9 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/input/EspressoButton/Tooltip.tsx]: Actually when I disabled that viewpoint hook, it raised TS error that `title` prop is required for AntD Tooltip

Comment (tn3rb) [reply assets/src/application/ui/input/EspressoButton/Tooltip.tsx]: that's bonkers:

```js
<DefaultTooltip title={tooltip} {...props}>
```
